vignettes/detection_efficiency_vignette.Rmd

```diff
@@ -0,0 +1,542 @@
+```{r load packages, message = FALSE}
+# ---- Bring in R packages ----
+{
+  library(dplyr)
+  library(ggplot2)
+  library(glatos)
+  library(mapview)
+  library(purrr)
+  library(sf)
+}
+```
+
+We will work with the example data set in [{glatos}](https://github.com/ocean-tracking-network/glatos) but the receiver deployment data either in table format or as sf object and a shapefile of the body of water you're working on. 
+```{r, results = 'hide'}
+#| title: get example data from glatos
+
+# get path to example receiver_locations file
+rec_file <- system.file("extdata",
+  "sample_receivers.csv",
+  package = "glatos"
+)
+
+# note that code above is needed to find the example file
+# for real glatos data, use something like below
+# rec_file <- "c:/path_to_file/GLATOS_receiverLocations_20150321_132242.csv"
+
+rcv <- read_glatos_receivers(rec_file)
+
+glimpse(rcv)
+```
+
+## Convert to `sf` object
+We will convert deployment data to a `sf` object and filter the data to only use the glatos array `"OSC"` for the example. You will not have to do the filtering when using your data but you will want to convert your receiver locations to a `sf` object and display using `mapview()`. 
+```{r, messages = FALSE}
+#| title: convert to sf and filter out array and specific station
+
+rcv_osc_sf <- rcv %>%
+  st_as_sf(
+    coords = c("deploy_long", "deploy_lat"),
+    crs = 4326
+  ) %>%
+  filter(glatos_array %in% "OSC")
+
+# view in mapview
+mapview(rcv_osc_sf)
+```
+
+We will filter out station number 12 as example to create our intial deployment rings. We will also convert into UTMs so we can estimate distances away from the receiver in meters. 
+```{r}
+#| title: filter out 12
+
+rcv_osc_sf_12 <- rcv_osc_sf %>%
+  filter(station_no %in% 12) %>%
+  st_transform(crs = 32617)
+```
+
+## Create buffer rings at set distances 
+Next, we will create buffer rings at our set distances away from the receiver. These distances are 100, 250, 500, and 750 but feel free to add more distances. If you add more distances, the model predictions will likely be more accurate. 
+
+```{r, warning = FALSE}
+#| title: create buffer rings
+
+# first create a data frame of distances to iterate over
+
+dists <- data.frame(
+  distance = c(100, 250, 500, 750)
+)
+# next we will split the data frame by distance and iterate over it using map
+buffer_rings <- dists %>%
+  split(.$distance) %>%
+  map(~ st_buffer(dist = .x$distance, rcv_osc_sf_12)) %>%
+  bind_rows(.id = "distance") %>%
+  st_cast("LINESTRING") %>%
+  dplyr::select(distance, glatos_array, station_no, ins_serial_no, geometry)
```

Comment:
This is a common warning with `st_cast()` when geometry lengths differ.

Reply:
I agree this is lower priority than other warnings I've flagged here (so OK releasing in this form with intent to clean up later), but readers not familiar with sf will appreciate you explaining to them what this means. In this particular case, this warning tells you that you are making an assumption that all geometries have the same attributes--there are consequences if this is not true. If we intend to show/teach best practices here, this should be cleaned up in the code (set attributes constant among geometries; or run the operation separately on each geometry) so that the warning isn't generated.